can/io/mf4.py

```diff
@@ -0,0 +1,160 @@
+# coding: utf-8
+
+"""
+Contains handling of MF4 logging files.
+
+"""
+
+from __future__ import absolute_import
+
+from datetime import datetime
+from pathlib import Path
+import logging
+
+from ..listener import Listener
+from ..util import channel2int
+from .generic import BaseIOHandler
+
+from asammdf import MDF, Signal
+import numpy as np
+
+CAN_MSG_EXT = 0x80000000
+CAN_ID_MASK = 0x1FFFFFFF
+
+DTYPE = np.dtype(
+    [
+        ('BusChannel', '<u1'),
+        ('ID', '<u4'),
+        ('IDE', '<u1'),
+        ('DLC', '<u1'),
+        ('DataLength', '<u1'),
+        ('DataBytes', '(64,)u1'),
+        ('Dir', '<u1'),
+        ('EDL', '<u1'),
+        ('BRS', '<u1'),
+        ('ESI', '<u1'),
+    ]
+)
+
+FD_LEN2DLC = {
+    12: 9,
+    16: 10,
+    20: 11,
+    24: 12,
+    32: 13,
+    48: 14,
+    64: 15,
+}
+
+logger = logging.getLogger('can.io.mf4')
+
+
+class MF4Writer(BaseIOHandler, Listener):
+    """Logs CAN data to an ASAM Measurement Data File v4 (.mf4).
+
+    If a message has a timestamp smaller than the previous one or None,
+    it gets assigned the timestamp that was written for the last message.
+    It the first message does not have a timestamp, it is set to zero.
+    """
+
+    def __init__(self, file, database=''):
```

Comment:
I would set `database=None` as a default, though your approach works.

```suggestion
    def __init__(self, file, database=None):
```

Nice docs, by the way!